File: source/dotnet/source/cmf.bus/cmf.rabbit/topology/Exchange.cs
﻿using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace cmf.rabbit.topology
{
    public struct Exchange
    {
        public string Name { get; private set; }
        public string HostName { get; private set; }
        public string VirtualHost { get; private set; }
        public int Port { get; private set; }
        public string RoutingKey { get; private set; }
        public string QueueName { get; private set; }
        public string ExchangeType { get; private set; }
        public bool IsDurable { get; private set; }
        public bool IsAutoDelete { get; private set; }
        public IDictionary Arguments { get; private set; }


        public Exchange(
            string name, 
            string hostName, 
            string vHost, 
            int port, 
            string routingKey, 
            string queueName = null, 
            string exchangeType = "topic", 
            bool isDurable = false, 
            bool autoDelete = false, 
            IDictionary arguments = null)
            : this()
        {
            this.Name = name;
            this.HostName = hostName;
            this.VirtualHost = vHost;
            this.Port = port;
            this.RoutingKey = routingKey;
            this.QueueName = queueName;
            this.ExchangeType = exchangeType;
            this.IsDurable = isDurable;
            this.IsAutoDelete = autoDelete;
            this.Arguments = arguments;
        }


        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("{");
            sb.AppendFormat("Name:{0},", this.Name);
            sb.AppendFormat("HostName:{0},", this.HostName);
            sb.AppendFormat("VirtualHost:{0},", this.VirtualHost);
            sb.AppendFormat("Port:{0},", this.Port);
            sb.AppendFormat("RoutingKey:{0},", this.RoutingKey);
            sb.AppendFormat("QueueName:{0},", this.QueueName);
            sb.AppendFormat("ExchangeType:{0},", this.ExchangeType);
            sb.AppendFormat("IsDurable:{0},", this.IsDurable);
            sb.AppendFormat("IsAutoDelete:{0},", this.IsAutoDelete);

            sb.Append("}");
            return sb.ToString();
        }
    }
}

File: source/java/src/main/java/cmf/rabbit/topology/Exchange.java
package cmf.rabbit.topology;

import java.util.Map;

public class Exchange {
	
	protected String name;
	protected String hostName;
	protected String virtualHost;
	protected int port;
	protected String routingKey;
	protected String queueName;
	protected String exchangeType;
	protected boolean isDurable;
	protected boolean isAutoDelete;
	@SuppressWarnings("rawtypes")
	protected Map arguments;
	
	
	public String getName() { return name; }
	protected void setName(String name) { this.name = name; }
	
	public String getHostName() { return hostName; }
	protected void setHostName(String hostName) { this.hostName = hostName; }
	
    public String getVirtualHost() { return virtualHost; }
    protected void setVirtualHost(String virtualHost) { this.virtualHost = virtualHost; }
    
    public int getPort() { return port; }
    protected void setPort(int port) { this.port = port; }
    
    public String getRoutingKey() { return routingKey; }
    protected void setRoutingKey(String routingKey) { this.routingKey = routingKey; }
    
    public String getQueueName() { return queueName; }
    protected void setQueueName(String queueName) { this.queueName = queueName; }
    
    public String getExchangeType() { return exchangeType; }
    protected void setExchangeType(String exchangeType) { this.exchangeType = exchangeType; }
    
    public boolean getIsDurable() { return isDurable; }
    protected void setIsDurable(boolean isDurable) { this.isDurable = isDurable; }
    
    public boolean getIsAutoDelete() { return isAutoDelete; }
    protected void setIsAutoDelete(boolean isAutoDelete) { this.isAutoDelete = isAutoDelete; }
    
    @SuppressWarnings("rawtypes")
	public Map getArguments() { return arguments; }
    @SuppressWarnings("rawtypes")
	protected void setArguments(Map arguments) { this.arguments = arguments; }


    @SuppressWarnings("rawtypes")
	public Exchange(
        String name, 
        String hostName, 
        String vHost, 
        int port, 
        String routingKey, 
        String queueName, 
        String exchangeType, 
        boolean isDurable, 
        boolean autoDelete, 
        Map arguments)
    {
        this.name = name;
        this.hostName = hostName;
        this.virtualHost = vHost;
        this.port = port;
        this.routingKey = routingKey;
        this.queueName = queueName;
        this.exchangeType = exchangeType;
        this.isDurable = isDurable;
        this.isAutoDelete = autoDelete;
        this.arguments = arguments;
    }

    
    @Override
    public String toString()
    {
        StringBuilder sb = new StringBuilder();

        sb.append("{");
        sb.append(String.format("Name:%s,", this.name));
        sb.append(String.format("HostName:{0},", this.hostName));
        sb.append(String.format("VirtualHost:{0},", this.virtualHost));
        sb.append(String.format("Port:{0},", this.port));
        sb.append(String.format("RoutingKey:{0},", this.routingKey));
        sb.append(String.format("QueueName:{0},", this.queueName));
        sb.append(String.format("ExchangeType:{0},", this.exchangeType));
        sb.append(String.format("IsDurable:{0},", this.isDurable));
        sb.append(String.format("IsAutoDelete:{0},", this.isAutoDelete));

        sb.append("}");
        return sb.toString();
    }
}
